Refs #25614 - Always expose the provider setting

ekohl opened the pull request:
This also adds integration tests for some modules.

ekohl committed:
Refs #25614 - Always expose the provider setting

ekohl committed:
Refs #25614 - Add integration tests for capabilities

Co-Authored-By: Tomer Brisker <tbrisker@gmail.com>

mmoll reviewed (dismissed):
Only checked visually, but looks correct.

tbrisker commented:
Pushed another commit with tests for other modules. logs and templates seem to work without the setting file expectation, seems like for some reason this method isn't called for them.

ekohl commented on test/realm_freeipa/integration_test.rb:
```diff
@@ -0,0 +1,30 @@
+require 'test_helper'
+require 'json'
+require 'root/root_v2_api'
+# require 'realm_freeipa/configuration_loader'
```
Could you remove the commented code here?

tbrisker replied:
oops, fixed

ekohl reviewed (commented):
I also considered testing whether it's running, but DHCP at least fails so I opted not to. If we can find an easy way to add why a plugin failed to the assertion that'd be a good test as well if we can fix it.

tbrisker commented:
[test core]

ekohl commented:
Could you fix the rubocop issues?

tbrisker commented:
:robot: :policeman:  fixed

tbrisker commented on modules/templates/templates_plugin.rb:
```diff
@@ -6,9 +6,5 @@ class Plugin < ::Proxy::Plugin
     validate_presence :template_url
-
-    after_activation do
-      loading_failed "missing :foreman_url: from configuration." unless Proxy::SETTINGS.foreman_url
```
This is actually required. TemplatesAPI creates a [TemplateProxyRequest](https://github.com/theforeman/smart-proxy/blob/develop/modules/templates/templates_api.rb#L19), which calls [`request_factory`](https://github.com/theforeman/smart-proxy/blob/develop/modules/templates/template_proxy_request.rb#L26) which [requires this setting](https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/request.rb#L57).

ekohl replied:
Ah, that does explain it.

tbrisker commented on test/templates/integration_test.rb:
```diff
@@ -0,0 +1,29 @@
+  end
+
+  def test_features
+    # not sure why, but this module loads without loading the config file
```
can this comment be removed then?

ekohl committed:
Refs #25614 - Fix typo in a comment

mmoll commented:
@ekohl which should go in first, GH-638 or this one?

ekohl commented:
Rebased and squashed. @tbrisker I've added you as `Co-Authored-By` on the integration tests. I think this is a cleaner history. Should be ready to be merged.

tbrisker reviewed (approved):
LGTM, but since i wrote part of the code i'll leave it to @mmoll to push the button

mmoll commented:
merged, bedankt @ekohl and thanks @tbrisker!
